### DISCUSSION TITLE
Redirect nanomq_cli output (messages) to file not working

### DISCUSSION THREAD

**Discussion by cocoon** [nanomq/nanomq#1018] [Q&A] [answered]
Hi,
I compiled nanomq and cli for an arm router and it works.

But when I try to redirect the output to a file, nothing appears in it. 
(using "nanomq_cli ... > file" or "nanomq_cli ... &> file" or additional with 2>&1)

On the console they appear without a problem.

I just wanted to subscribe a specific topic and record the incoming messages. 

Is there a simple way to do it without modifying code? I just searched and tried, even with nohup (tee not available).

I thought I will ask while searching for a way to get it working, maybe I just oversee something obvious.

For example a loglevel messages would help, maybe I will try to figure out this directly in nanomq, then I will directly save messages from there instead of using the cli in addition.

If anybody has an idea or solution let me know.

#######

Update:
it looks like that a buffer is filled and for me currently around after 30 minutes it gets flushed and then ends up in the file.

Update2:
It is not always the same amount of time, currently in the logfile the last entry is from:
2023-01-18T11:57:31

but in my other MQTT Client I see the last message (every 5 minutes) is from:
2023-01-18T12:39:53

One more indicator that a buffer gets flushed is, that in the file don't appear full lines, the last line is cut off most of the time.

I will try to try the same on a real full Linux VM how it behaves there.

**Comment by JaylinYu** [comment]
This deserve an issue. nanomq_cli is still on preliminary stage(lacking of verbose output),will complete it in next version. 
But didn't find similar problem as yours.

**Comment by JaylinYu** [comment, marked as answer]
It has been identified as a bug, @alvin1221  fixed this in the master branch.

**Reply by cocoon** [reply]
Wow thanks that was quick, will try! 

Update: yes, looks much better now, text instantly appears in the file, great work!
